core/store/src/trie/trie_storage.rs

```diff
@@ -176,6 +176,7 @@ impl TrieStorage for TrieCachingStorage {
                 .get(ColState, key.as_ref())
                 .map_err(|_| StorageError::StorageInternalError)?;
             if let Some(val) = val {
+                let val: Arc<[u8]> = val.into();
```

Comment:
In the grand scheme on things, this copy is avoidable. Here's what happens here:

The data lives in memory, in some internal rocks datastructure. We then copy the data from rocks into a Rust Vec (there's an alternative API to get an equivalent of `[u8]`, but we don't use that). And then we copy data from Vec into Arc. 

The optimization here would be to elide intermediete copy, and to copy data from rocks directly into an Arc.

I don't think we need to do this in the PR, but it's something to keep in mind when thinking about storage.